src/agent/agent.c

```diff
@@ -32,6 +32,8 @@
 #define DEBUG_SYSTEMD_MESSAGES 0
 #define DEBUG_SYSTEMD_MESSAGES_CONTENT 0
 
+#define CONNECTION_RETRY_COUNT_UNTIL_QUIET 10
```

Comment:
Maybe make it configurable in `agent.conf`?

Reply:
I thought about it and wasn't sure its worth it... but its probably better to do it now.

Reply:
Added :heavy_check_mark: 
@mkemel PTAL